test/amp_tags_test.js

```diff
@@ -13,46 +13,52 @@ describe('AmpTags', function() {
-    const string = getSeoMetadata(seoConfig, config, 'home-page', {data: {}}, {})
+  it("does not ampify other pages", function () {
+    const string = getSeoMetadata(seoConfig, config, 'home-page', { data: {} }, {})
     assert.equal('', string);
   });
 
   it("does allows you to only ampify free story pages", function () {
     const story = { "slug": "section/slug", "is-amp-supported": true }
-    const publicStoryResults = getSeoMetadata({...seoConfig, ampStoryPages: "public"}, config, 'story-page', { data: { story: story } }, {})
+    const publicStoryResults = getSeoMetadata({ ...seoConfig, ampStoryPages: "public" }, config, 'story-page', { data: { story: story } }, {})
     assert.equal('<link rel="amphtml" href="/amp/story/section%2Fslug"/>', publicStoryResults);
-    const privateStoryResults = getSeoMetadata({ ...seoConfig, ampStoryPages: "public" }, config, 'story-page', { data: { story: {...story, access: "subscription"} } }, {})
+    const privateStoryResults = getSeoMetadata({ ...seoConfig, ampStoryPages: "public" }, config, 'story-page', { data: { story: { ...story, access: "subscription" } } }, {})
     assert.equal('', privateStoryResults);
   })
-  
+
   it("does not append domain to amp stories if appendHostToAmpUrl not present", function () {
-    const story = {"slug": "section/slug", "is-amp-supported": true}
-    const string = getSeoMetadata(seoConfig, config, 'story-page', {data: {story: story}}, {})
+    const story = { "slug": "section/slug", "is-amp-supported": true }
+    const string = getSeoMetadata(seoConfig, config, 'story-page', { data: { story: story } }, {})
     assertContains('<link rel="amphtml" href="/amp/story/section%2Fslug"/>', string);
   })
 
-  it("does append domain to amp stories if appendHostToAmpUrl present", function () {
-    const story = {"slug": "section/slug", "is-amp-supported": true}
-    const string = getSeoMetadata({...seoConfig, appendHostToAmpUrl: true}, config, 'story-page', {currentHostUrl: "https://madrid.quintype.io/section/slug", data: {story}}, {})
+  it("does append domain to amp stories if appendHostToAmpUrl and domainSlug present", function () {
+    const story = { "slug": "section/slug", "is-amp-supported": true }
+    const string = getSeoMetadata({ ...seoConfig, appendHostToAmpUrl: true }, config, 'story-page', { currentHostUrl: "https://news.madrid.quintype.io/section/slug", domainSlug: 'news', data: { story } }, {})
+    assertContains('<link rel="amphtml" href="https://news.madrid.quintype.io/amp/story/section%2Fslug"/>', string);
+  })
+  it("does append domain to amp stories if appendHostToAmpUrl and domainSlug is not present", function () {
+    const story = { "slug": "section/slug", "is-amp-supported": true }
+    const string = getSeoMetadata({ ...seoConfig, appendHostToAmpUrl: true }, config, 'story-page', { currentHostUrl: "https://news.madrid.quintype.io/section/slug", domainSlug: undefined, data: { story } }, {})
```

Comment:
As per my understanding, for this test case u should remove `news` from `https://news.madrid.quintype.io/section/slug` and remove `domainSlug` key itself instead of passing undefined.